packages/x-date-pickers/src/internals/hooks/useField/useField.utils.ts

```diff
@@ -408,12 +405,12 @@ export const doesSectionFormatHaveLeadingZeros = <TDate extends PickerValidDate>
     return false;
   }
 
-  const now = utils.date(undefined, timezone);
+  const now = utils.date(undefined, 'default');
```

Comment:
I noticed that `useParsedFormat` didn't really need the `timezone`, all the utils that it uses required the timezone but did not really use it (when checking if a section format has a leading zero, using the right timezone or the default one changes nothing).

So I removed the `timezone` param where it was not needed.
I will do a 2nd pass after this PR to see if others utils don't need it, but it's not related to this PR anymore.

And why did those utils had the timezone in the first place?
Well when I added the feature I just added it everywhere for safety but now I know it's pointless :laughing:

Reply:
Nitpick: it could make sense to have such changes in a separate PR. 🤔

Reply:
If you are ok with it, I'll merge it inside this one to not spend more time on this PR :angel:

Reply:
Sure, not a problem. 👍